cms/djangoapps/appsembler/views.py

```diff
@@ -274,3 +281,39 @@ def log_multiple_objects_returned(self):
     def render_login_page_with_error(self, error_code):
         return render_login_page(
             login_error_message=self.error_messages[error_code])
+
+
+def get_logout_redirect_url(request):
+    """
+    Return logout redirect url using the site related to the given user if possible.
+    Otherwise, return settings.LOGOUT_REDIRECT_URL
+
+    :return: logout redirect url or settings.LOGOUT_REDIRECT_URL
+    """
+    user = getattr(request, 'user', None)
+    if not user or not user.is_authenticated:
+        return reverse(settings.LOGOUT_REDIRECT_URL)
+
+    try:
+        organization = get_organization_for_user(user=user)
+    except ObjectDoesNotExist:
+        return reverse(settings.LOGOUT_REDIRECT_URL)
+    site = get_site_by_organization(organization=organization)
+
+    return '{protocol}://{site_domain}'.format(
+        protocol='https' if request.is_secure() else 'http',
+        site_domain=site.domain
+    )
+
+
+class StudioLogoutView(View):
+    """
+    Studio Logout View
+    """
+    @method_decorator(csrf_protect)
+    @method_decorator(login_required)
+    def get(self, request):
+        """
+        Perform logout from studio, and redirect to LMS home page
+        """
+        return DjangoLogoutView.as_view(next_page=get_logout_redirect_url(request))(request)
```

Comment:
I think that's a more maintainable implementation that using `.as_view`.

```suggestion
class StudioLogoutView(DjangoLogoutView):
    @property
    def next_page(self):
        return get_logout_redirect_url(self.request)
```

Reply:
hahaha, stupid me,,, Thank you @OmarIthawi

Reply:
@OmarIthawi, it didn't work, because the `request` is losing the `user` by `auth_logout` before fetching `next_page`. See [django code here](https://github.com/django/django/blob/bf07047f45767eddaf52400c60f7e2becc8891d3/django/contrib/auth/views.py#L116)

The possible workaround with inheritance is uglier than my code. I'm keeping the code as it is unless you have another idea 😁

Reply:
no problem. let's keep it as-is